From 10f25506f65b7d2108bbbaf75169c8d0ee732e0a Mon Sep 17 00:00:00 2001
From: Vineesh k k <vineesh.k.k@intel.com>
Date: Mon, 25 May 2015 14:39:11 +0530
Subject: [PATCH] EM/NX2A4WP: Klocwork issue needs to be fixed on NX2A4WP

Fix overflow issue when the ptu_power_code array index is > 5

Change-Id: Iaeb9f30c16dcee73d0d7bdd011fa7dbc5c5704ca
Tracked-On: https://jira01.devtools.intel.com/browse/AKP-1708
Signed-off-by: Vineesh k k <vineesh.k.k@intel.com>
---
 drivers/power/wireless_nx2a4wp.c | 25 +++++++++++++------------
 1 file changed, 13 insertions(+), 12 deletions(-)

diff --git a/drivers/power/wireless_nx2a4wp.c b/drivers/power/wireless_nx2a4wp.c
index b9cce0c81ff9..50cf852aa3a5 100644
--- a/drivers/power/wireless_nx2a4wp.c
+++ b/drivers/power/wireless_nx2a4wp.c
@@ -127,6 +127,8 @@
 #define A4WP_LONG_BEACON_TIME	(100 * HZ/1000)
 
 #define PRU_MAX_MA	900
+#define PTU_DEFAULT_POWER	5500
+
 
 #define IRECT_BIT_RESOLUTION_INV	1715
 #define IRECT_ADC12_TO_MA(__value)  \
@@ -799,7 +801,12 @@ static int nx2a4wp_usb_set_property(struct power_supply *psy,
 	case POWER_SUPPLY_PROP_PTU_POWER:
 		temp = find_adc_code(val->intval,
 			PTU_POWER_TABLE_SIZE, ptu_power_code);
-		chip->ptu_power = ptu_power_code[temp][1];
+		if (temp >= 0)
+			chip->ptu_power =
+				ptu_power_code[temp][1];
+		else
+			chip->ptu_power =
+				PTU_DEFAULT_POWER;
 		break;
 	case POWER_SUPPLY_PROP_PTU_CLASS:
 		chip->ptu_class = val->intval;
@@ -952,9 +959,7 @@ static irqreturn_t nx2a4wp_thread_handler(int id, void *data)
 		if (!test_bit(A4WP_FLAGS_PRESENT_BIT, &chip->flags)) {
 			chip->present_timestamp = chip->irq_timestamp;
 			nx2a4wp_evt_present_lock(chip);
-		}
-		/* Health event */
-		else {
+		} else {
 			int ret;
 
 			/* No more Interrupt */
@@ -974,10 +979,8 @@ static irqreturn_t nx2a4wp_thread_handler(int id, void *data)
 				power_supply_changed(&chip->psy_wc);
 			}
 		}
-	}
-
-	/* Falling Edge */
-	else {
+	} else {
+		/* Falling Edge */
 		int ret1, ret2;
 
 		ret1 = nx2a4wp_read_reg(chip->client, NX2A4WP_STATL_ADDR);
@@ -989,10 +992,8 @@ static irqreturn_t nx2a4wp_thread_handler(int id, void *data)
 		if ((ret1 < 0) || (ret2 < 0)) {
 			if (test_bit(A4WP_FLAGS_PRESENT_BIT, &chip->flags))
 				nx2a4wp_evt_nopresent_lock(chip);
-		}
-
-		/* Health event */
-		else {
+		} else {
+			/* Health event */
 			unsigned int health;
 			if (ret2 & NX2A4WP_OVP_MASK)
 				health = POWER_SUPPLY_HEALTH_OVERVOLTAGE;
-- 
1.9.1

